include/inexor/vulkan-renderer/wrapper/device.hpp

```diff
@@ -14,8 +14,77 @@
+
+/// Call vkGetPhysicalDeviceMemoryProperties
+/// @param physical_device The physical device
+/// @return The memory properties of the physical device
+[[nodiscard]] VkPhysicalDeviceMemoryProperties get_physical_device_memory_properties(VkPhysicalDevice physical_device);
+
+/// Call vkGetPhysicalDeviceProperties
+/// @param physical_device The physical device
+/// @return The user-friendly name of the physical device
+[[nodiscard]] std::string get_physical_device_name(VkPhysicalDevice physical_device);
+
+/// Call vkGetPhysicalDeviceProperties
+/// @param physical_device The physical device
+/// @return The physical device type
+[[nodiscard]] VkPhysicalDeviceType get_physical_device_type(VkPhysicalDevice physical_device);
+
+/// Check if a certain device extension is available for a specific graphics card
+/// @param physical_device The physical device
+/// @param extension The name of the device extension
+/// @return ``true`` if the requested device extension is available
+[[nodiscard]] bool is_extension_supported(VkPhysicalDevice physical_device, const std::string &extension);
+
+/// Call vkGetPhysicalDeviceSurfaceSupportKHR
+/// @param physical_device The physical device
+/// @param surface The window surface
+/// @return ``true`` if presentation is supported
+[[nodiscard]] bool is_presentation_supported(VkPhysicalDevice physical_device, VkSurfaceKHR surface);
+
+/// Call is_extension_supported with ``VK_KHR_SWAPCHAIN_EXTENSION_NAME``
+/// @param physical_device The physical device
+/// @return ``true`` if swapchain is supported
+[[nodiscard]] bool is_swapchain_supported(VkPhysicalDevice physical_device);
+
+/// Automatically select the best physical device (graphics card) out of all the available ones
+/// The user can manually specify which graphics card will be used with command line argument ``--gpu <index>``
+/// The method will check if the specified index is a valid array index
+/// Please note that the graphics cards index starts with 0!
+/// @param inst The Vulkan instance
+/// @param surface The window surface
+/// @param preferred_index The preferred graphics card index (starting with ``0``!)
+/// @return The selected physical device (if any could be found), or std::nullopt otherwise
+[[nodiscard]] std::optional<VkPhysicalDevice>
+pick_graphics_card(VkInstance inst, VkSurfaceKHR surface, std::optional<std::uint32_t> preferred_index = std::nullopt);
+
+/// Rate a graphics card by its features and properties
+/// @param type The physical device type
+/// @param memory_props The memory properties of the physical device
+/// @param swapchain_supported ``true`` if the physical device supports swapchains
+/// @param presentation_supported ``true`` if the physical device supports presentation
+/// @return The graphics card's score
+/// @note If the score is smaller than ``0``, this means the physical device is unsuitable and can't be used!
+[[nodiscard]] std::int64_t rate_physical_device(VkPhysicalDeviceType type,
```

Comment:
Good point! This also fixes the clang-tidy warning.

Reply:
I removed this method again. When picking a physical device automatically, I filter out all physical devices which are unsuitable und then rank them by type first, and then by total memory score second in case the type is the same.